### PR TITLE
Add ACH support

### PR DESCRIPTION
**Pull request by ritchiea**

### CODE REVIEW

File: lib/stripe/ext/active_merchant/active_merchant.rb [hunk 1]
```diff
@@ -37,6 +41,118 @@ def headers(options = {})
...
+        end.responses.last
+      end
+
+      def store(payment, options = {})
+        params = {}
+        post = {}
+
+        if payment.is_a?(ApplePayPaymentToken)
+          token_exchange_response = tokenize_apple_pay_token(payment)
+          params = { card: token_exchange_response.params["token"]["id"] } if token_exchange_response.success?
```
Comment (pierre) [inline lib/stripe/ext/active_merchant/active_merchant.rb]: Ruby 2.0 syntax won't work by default, better to stick with the old syntax for now.

File: spec/stripe/remote/integration_spec.rb [hunk 2]
```diff
@@ -190,4 +188,54 @@
...
+    pm.country.should eq("US")
+    pm.source_type.should eq("bank_account")
+
+    properties = [build_property(:token, pm.token)]
+    kb_payment = @plugin.kb_apis.proxied_services[:payment_api].add_payment(SecureRandom.uuid)
+
+    attempt_response = @plugin.purchase_payment(@pm.kb_account_id, kb_payment.id, kb_payment.transactions[0].id, pm.kb_payment_method_id, @amount, @currency, properties, @call_context)
+    attempt_response.status.should eq(:ERROR), attempt_response.gateway_error
+
+    @plugin.verify_bank_account(pm.stripe_customer_id, pm.token, bank_account_verification_numbers, @call_context.tenant_id)
```
Comment (pierre) [inline spec/stripe/remote/integration_spec.rb]: Is there a way to check the verification was successful? The test is failing for me (wondering if there's a delay?):

```
Failures:

  1) Killbill::Stripe::PaymentPlugin should be able to add a bank account, verify and accept a bank charge
     Failure/Error: payment_response.status.should eq(:PROCESSED), payment_response.gateway_error
       The customer's bank account must be verified in order to create an ACH payment. For more information, see https://stripe.com/docs/guides/ach#manually-collecting-and-verifying-bank-accounts
       Diff:
       @@ -1,2 +1,2 @@
       -:PROCESSED
       +:ERROR

     # ./spec/stripe/remote/integration_spec.rb:211:in `(root)'
```

Comment (ritchiea) [reply spec/stripe/remote/integration_spec.rb]: That test ended up failing for me as well but I couldn't figure out a good reason for the failure so I thought I'd open the PR and see what happened for another developer.

I have been using the [Stripe Ruby client](https://github.com/stripe/stripe-ruby) to check the verification. If you use the same API key and query the bank account using it's token after `verify_bank_account` here the Stripe API returns that the card is verified. There is a `status` property for the card that is set to `verified` only after verification. So I'm not sure what the issue is, the error returned from Stripe does not fit with the other data being returned from their API calls. Maybe it is a delay? Maybe I should reach out to Stripe?

File: lib/stripe/ext/active_merchant/active_merchant.rb (hunk 1 above)
Comment (ritchiea) [reply lib/stripe/ext/active_merchant/active_merchant.rb]: Got it, easy fix

File: lib/stripe/api.rb [hunk 3]
```diff
@@ -246,6 +265,21 @@ def get_application_fee(amount, properties)
 
         config(context.tenant_id)[:stripe][:fees_amount] || (config(context.tenant_id)[:stripe][:fees_percent].to_f * amount * 100)
       end
+
+      def is_bank_account?(properties)
+        find_value_from_properties(properties, :routing_number) &&
```
[the remark this answers is not in the document]
Comment (ritchiea) [reply lib/stripe/api.rb]: Will do

File: db/schema.rb [hunk 4]
```diff
@@ -5,6 +5,11 @@
     t.string   "kb_payment_method_id"      # NULL before Kill Bill knows about it
     t.string   "token"                     # stripe id
     t.string   "stripe_customer_id"
+    t.string   "source_type"
```
[the remark this answers is not in the document]
Comment (ritchiea) [reply db/schema.rb]: Just a standard ActiveRecord migration? I wasn't sure if the plugin required migrations or could load it's table from the schema file.

File: lib/stripe/ext/active_merchant/active_merchant.rb [hunk 5]
```diff
@@ -37,6 +41,118 @@ def headers(options = {})
...
+        elsif payment.is_a?(KillBill::Stripe::BankAccount)
+          post.merge!(bank_account_params(payment, options))
+        else
+          add_creditcard(params, payment, options)
+        end
+
+        unless payment.is_a?(KillBill::Stripe::BankAccount)
```
[the remark this answers is not in the document]
Comment (ritchiea) [reply lib/stripe/ext/active_merchant/active_merchant.rb]: Good idea, I'll go ahead with that.

File: db/schema.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (pierre) [reply db/schema.rb]: Did you go through the doc linked?

File: spec/stripe/remote/integration_spec.rb (hunk 2 above)
Comment (pierre) [reply spec/stripe/remote/integration_spec.rb]: I would indeed reach out to Stripe to understand if there is a delay.

File: db/schema.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (ritchiea) [reply db/schema.rb]: Yea I read through the doc and it looks to me like all I need is an ActiveRecord migration. Is that correct or am I misunderstanding something?

[the remark this answers is not in the document]
Comment (pierre) [reply db/schema.rb]: Pretty much, see https://github.com/killbill/killbill-stripe-plugin/tree/master/db/migrate for an example.

**Comment by pierre** [comment]
All good, thanks!

**Pull request merged**